core/src/test/java/org/hibernate/ogm/backendtck/queries/pagination/PaginationUseCases.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.backendtck.queries.pagination;
+
+import static java.util.Arrays.copyOfRange;
+import static org.fest.assertions.Assertions.assertThat;
+
+import java.util.List;
+
+import org.hibernate.Session;
+import org.hibernate.ogm.utils.OgmTestCase;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+/**
+ * Test the pagination for native queries with MongoDB
```

Comment:
this test is now generic, maybe we could remove the reference to the dialect

Reply:
Updated and pushed force